File: Assets/My Assets/Scripts/Level/generateField.js
var asteroids : int = 200;
var maxScale : float = 3;
var spawnRange : int = 10000;
var myAsteroid : GameObject;
function Start(){
	Random.seed = System.DateTime.Now.Ticks;
	generateField();
}
function generateField(){ //Goal: generate an asteroid field.
	for(var i = 0; i < asteroids; i++){
		var newAsteroid = Instantiate(myAsteroid,this.transform.position,Quaternion.identity);
		newAsteroid.transform.position = Vector3(Random.Range(-spawnRange,spawnRange),Random.Range(-spawnRange,spawnRange),0);
		var myScale = Random.Range(0.26,maxScale);
		newAsteroid.transform.localScale.x = myScale;
		newAsteroid.transform.localScale.y = myScale;
	}
}

File: Assets/My Assets/Scripts/Level/generateField.cs
using UnityEngine;
using System.Collections;

[System.Serializable]
public class generateField : MonoBehaviour
{
    public int asteroids;
    public float maxScale;
    public int spawnRange;
    public GameObject myAsteroid;
    public virtual void Start()
    {
        Random.seed = (int) System.DateTime.Now.Ticks;
        this.generateMyFields();
    }

    public virtual void generateMyFields() //Goal: generate an asteroid field.
    {
        int i = 0;
        while (i < this.asteroids)
        {
            GameObject newAsteroid = UnityEngine.Object.Instantiate(this.myAsteroid, this.transform.position, Quaternion.identity);
            newAsteroid.transform.position = new Vector3(Random.Range(-this.spawnRange, this.spawnRange), Random.Range(-this.spawnRange, this.spawnRange), 0);
            float myScale = Random.Range(0.26f, this.maxScale);
            Vector3 newRoid = newAsteroid.transform.localScale;
            newRoid.x = myScale;
            newRoid.y = myScale;
            newAsteroid.transform.localScale = newRoid;
            i++;
        }
    }

    public  bool generateFields()
    {
        this.asteroids = 200;
        this.maxScale = 3;
        this.spawnRange = 10000;
        return true;
    }

}